File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-12/05-canPlaceFlowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0: return True
        if len(flowerbed) == 1:
            return flowerbed[0] == 0
        
        count = 0
        i = 0
        while i < len(flowerbed):
            if flowerbed[i] == 0:
                if i == 0:
                    if flowerbed[i+1] == 0:
                        count += 1
                        i += 1
                elif i == len(flowerbed) - 1:
                    if flowerbed[i-1] == 0:
                        count += 1
                        i += 1
                else:
                    if flowerbed[i-1] == 0 and flowerbed[i+1] == 0:
                        count += 1
                        i += 1
            i += 1
            
            if count == n:
                return True
        return False

File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-12/05-canPlaceFlowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        if (n == 0) return true;
        if (flowerbed.length == 1)
            return flowerbed[0] == 0;
        
        int count = 0;
        int i = 0;
        while (i < flowerbed.length) {
            if (flowerbed[i] == 0) {
                if (i == 0) {
                    if (flowerbed[i+1] == 0) {
                        count++;
                        i++;
                    }
                } else if (i == flowerbed.length - 1) {
                    if (flowerbed[i-1] == 0) {
                        count++;
                        i++;
                    }
                } else {
                    if (flowerbed[i-1] == 0 && flowerbed[i+1] == 0) {
                        count++;
                        i++;
                    }
                }
            }
            i++;
            if (count == n) return true;
        }
        return false;
    }
}
